DigitalLearningSolutions.Data/Services/NotificationService.cs

```diff
@@ -1,61 +1,94 @@
-        }
-    }
-}
+﻿namespace DigitalLearningSolutions.Data.Services
+{
+    using System;
+    using System.Threading.Tasks;
+    using DigitalLearningSolutions.Data.Helpers;
+    using DigitalLearningSolutions.Data.Models;
+    using DigitalLearningSolutions.Data.Models.Email;
+    using Microsoft.Extensions.Configuration;
+    using Microsoft.FeatureManagement;
+    using MimeKit;
+
+    public interface INotificationService
+    {
+        Task SendUnlockRequest(int progressId);
+    }
+
+    public class NotificationService : INotificationService
+    {
+        private readonly IConfigService configService;
+        private readonly IConfiguration configuration;
+        private readonly IEmailService emailService;
+        private readonly IFeatureManager featureManager;
+        private readonly INotificationDataService notificationDataService;
+
+        public NotificationService(
+            IConfiguration configuration,
+            INotificationDataService notificationDataService,
+            IConfigService configService,
+            IEmailService emailService,
+            IFeatureManager featureManager
+        )
+        {
+            this.configuration = configuration;
+            this.notificationDataService = notificationDataService;
+            this.configService = configService;
+            this.emailService = emailService;
+            this.featureManager = featureManager;
+        }
+
+        public async Task SendUnlockRequest(int progressId)
+        {
+            var unlockData = notificationDataService.GetUnlockData(progressId);
+            if (unlockData == null)
+            {
+                throw new NotificationDataException(
+                    $"No record found when trying to fetch Unlock Data. Progress ID: {progressId}"
+                );
+            }
+
+            unlockData.ContactForename =
+                unlockData.ContactForename == "" ? "Colleague" : unlockData.ContactForename;
+            var refactoredTrackingSystemEnabled = await featureManager.IsEnabledAsync("RefactoredTrackingSystem");
+
+            var baseUrlConfigOption = refactoredTrackingSystemEnabled
+                ? configuration.GetAppRootPath()
+                : configService.GetConfigValue(ConfigHelper.CurrentSystemBaseUrlName);
+            if (baseUrlConfigOption == null)
+            {
+                throw new ConfigValueMissingException(
+                    configService.GetConfigValueMissingExceptionMessage(
+                        refactoredTrackingSystemEnabled ? "AppRootPath" : "CurrentSystemBaseUrl"
+                    )
+                );
+            }
+
+            var baseUrl = refactoredTrackingSystemEnabled
+                ? $"{configuration.GetAppRootPath()}/TrackingSystem/Delegates/CourseDelegates"
+                : $"{configuration.GetCurrentSystemBaseUrl()}/Tracking/";

```

Comment:
I don't think this will create the correct old site url, there is no coursedelegates in the path any more. With the feature off, the code needs to do the same as it did previously.